File: angularclient/src/app/model/wallbox.ts
import { WallboxOwner } from "./owner";

export interface Wallbox {
    id: number;
    physicalId: number;
    name: string;
    owner: WallboxOwner
}

export interface RentedWallbox {
    id: number;
    wallbox: Wallbox;
    startTime: Date;
    endTime: Date;
}

File: src/main/java/com/datix/coresystem_poc/entity/Wallbox.java
package com.datix.coresystem_poc.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED) // Required by JPA
@AllArgsConstructor
@Builder
@ToString
public class Wallbox {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    private String physicalId;

    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    private WallboxOwner owner;

}
